src/gui/statsdialog.cpp

```diff
@@ -90,21 +90,13 @@ void StatsDialog::update()
     // From lt manual: disk_write_queue and disk_read_queue are the number of peers currently waiting on a disk write or disk read
     // to complete before it receives or sends any more data on the socket. It's a metric of how disk bound you are.
 
-    // num_peers is not reliable (adds up peers, which didn't even overcome tcp handshake)
-    const auto torrents = BitTorrent::Session::instance()->torrents();
-    const quint32 peers = std::accumulate(torrents.cbegin(), torrents.cend(), 0, [](const quint32 acc, const BitTorrent::TorrentHandle *torrent)
-    {
-        return (acc + torrent->peersCount());
-    });
+    m_ui->labelWriteStarve->setText(QString("%1%").arg(((ss.diskWriteQueue > 0) && (ss.peersCount > 0))
+        ? Utils::String::fromDouble((100. * ss.diskWriteQueue / ss.peersCount), 2)
+        : QLatin1String("0")));
+    m_ui->labelReadStarve->setText(QString("%1%").arg(((ss.diskReadQueue > 0) && (ss.peersCount > 0))
+        ? Utils::String::fromDouble((100. * ss.diskReadQueue) / ss.peersCount, 2)
```

Comment:
Notice the parentheses placement is now different than in L94 and I'm in favor of using the L94 form. And synccontroller.cpp should be kept in sync as well.

Reply:
Good catch, fixed. `synccontroller.cpp` was already using the correct form.

Reply:
I think there is some misunderstanding here, I would prefer writing `(100. * ss.diskWriteQueue / ss.peersCount)` instead of the other one. I think most of the time expressions should be properly enclosed by parentheses.

Reply:
Ok yeah I thought you meant the other way around. Fixed.